wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/SyncUsersActionHandler.swift

```diff
@@ -0,0 +1,124 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireDataModel
+
+// rename to describe v4
+class SyncUsersActionHandler: ActionHandler<SyncUsersAction> {
+
+    private let userProfileProcessor = UserProfilePayloadProcessor()
+
+    // MARK: - Request
+
+    struct RequestPayload: Codable, Equatable {
+
+        let qualified_users: [QualifiedID]
+
+    }
+
+    private func markUserProfilesAsUnavailable(_ users: Set<QualifiedID>) {
+        for qualifiedID in users {
+            let user = ZMUser.fetch(with: qualifiedID.uuid, domain: qualifiedID.domain, in: context)
+            user?.isPendingMetadataRefresh = true
+            user?.needsToBeUpdatedFromBackend = false
+        }
+    }
```

Comment:
You should probably do this in `context.performAndWait`.

Reply:
Done!